File: educapais_flutter/lib/services/api_service.dart
import 'dart:convert';

import 'package:http/http.dart' as http;

class ApiService {
  ApiService({http.Client? client}) : _client = client ?? http.Client();

  static const endpoint =
      'https://69effdd0112e1b968e251fe2.mockapi.io/api/educapais';

  final http.Client _client;

  Future<String> fetchPayload() async {
    final uri = Uri.parse(endpoint);
    final response = await _client
        .get(uri, headers: {'accept': 'application/json'})
        .timeout(const Duration(seconds: 12));

    if (response.statusCode < 200 || response.statusCode >= 300) {
      throw Exception('Erro HTTP ${response.statusCode}');
    }

    return utf8.decode(response.bodyBytes);
  }
}

File: educapais_ios/EducaPais/Services/APIService.swift
import Foundation

struct APIService {
    static let endpoint = URL(string: "https://69effdd0112e1b968e251fe2.mockapi.io/api/educapais")!

    func fetchPayload() async throws -> String {
        let (data, response) = try await URLSession.shared.data(from: Self.endpoint)
        guard let httpResponse = response as? HTTPURLResponse,
              (200 ..< 300).contains(httpResponse.statusCode) else {
            throw URLError(.badServerResponse)
        }

        if let text = String(data: data, encoding: .utf8) {
            return text
        }
        if let text = String(data: data, encoding: .isoLatin1) {
            return text
        }
        throw URLError(.cannotDecodeRawData)
    }
}
